src/solc_0.8/Game/GameBaseToken.sol

```diff
@@ -87,8 +104,15 @@ contract GameBaseToken is ImmutableERC721, WithMinter, Initializable, IGameToken
         if (editor != address(0)) {
             _setGameEditor(to, editor, true);
         }
-        if (creation.assetIdsToAdd.length != 0) {
-            _addAssets(from, strgId, creation.assetIdsToAdd, creation.assetAmountsToAdd);
+
+        GameData721 memory creation721 = creation.gameData721;
+        if (creation721.assetIdsToAdd.length != 0) {
```

Comment:
Is it not possible to do it without initializing a new variable ?
```
if (creation.gameData721.assetIdsToAdd.length != 0) {
```

Also, ```gameDataERC721```

Reply:
Yes we can do it.
We will spend less gaz but the code will be a bit less clean ton me.
Let's do it